### PR TITLE
Adds Query API

### PR DESCRIPTION
**Pull request by carsonfarmer**
Also comes with a few minor nits and fixes. But primarily, ports @jsign's Query API over to JS, so we can do things like:

```javascript
interface Person {
  ID: string
  age: number
}
...
const people = [...Array(8)].map((_, i) => {
  return { ID: ""+i, age: 60 + i }
})
await client.modelCreate<Person>(store.id, 'Person', people)
const q = new Where('age')
  .ge(60)
  .and('age')
  .lt(66)
  .or(new Where('age').eq(67))
const find = await client.modelFind<Person>(store.id, 'Person', q)
```

It also supports a slightly more explicit API style via a static method on Query:

```javascript
const q = Query.where('age')
  .ge(60)
  .and('age')
  .lt(66)
  .or(new Where('age').eq(67))
```

### CODE REVIEW

**Comment by carsonfarmer** [comment]
Note that I haven't (yet) exposed the Query API directly on the Client...
Depending on reviews, we _could_ add a method like: `client.FindWhere(fieldPath: string)...` or something. But for now, just leaving the Query building API on its own...

**Comment by carsonfarmer** [comment]
Turned off tests for now... don't know why I didn't think of that sooner 😉

**Comment by jsign** [comment]
Looks good and familiar! :wink:

File: src/query.ts [hunk 1]
```diff
@@ -1,44 +1,123 @@
...
+
+  // OrderByDesc specify descending order for the query results.
+  // On multiple calls, only the last one is considered.
+  orderByDesc(fieldPath: string): Query {
+    this.sort = { fieldPath, desc: true }
+    return this
+  }
 }
+
+// Sanity check
```
Comment (sanderpick) [inline src/query.ts]: Retain?

File: src/index.ts [hunk 2]
```diff
@@ -34,15 +29,14 @@ import { fromBase64, toBase64 } from 'b64-lite'
 import * as pack from '../package.json'
 import { ReadTransaction } from './ReadTransaction'
 import { WriteTransaction } from './WriteTransaction'
-import { JSONQuery } from './query'
+import { JSONQuery, Entity, EntityList } from './models'
```
Comment (sanderpick) [inline src/index.ts]: One thing I noticed from playing around yesterday is the need to expose more things other than `Client`, like `JSONQuery`. Probably other stuff too.

Comment (carsonfarmer) [reply src/index.ts]: Ah yeh good call! I'll those in before merging.

File: src/query.ts (hunk 1 above)
Comment (carsonfarmer) [reply src/query.ts]: Yeh, probs not.

**Pull request merged**